File: remote/src/main/java/com/kelsos/mbrc/presenters/PlaylistDialogPresenterImpl.java
package com.kelsos.mbrc.presenters;

import com.google.inject.Inject;
import com.kelsos.mbrc.interactors.PlaylistInteractor;
import com.kelsos.mbrc.ui.views.PlaylistDialogView;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;
import timber.log.Timber;

public class PlaylistDialogPresenterImpl implements PlaylistDialogPresenter {
  private PlaylistDialogView view;

  @Inject private PlaylistInteractor playlistInteractor;

  @Override public void load() {
    playlistInteractor.getUserPlaylists()
        .subscribeOn(Schedulers.io())
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe(playlists -> {
          view.update(playlists);
        }, throwable -> {
          Timber.v(throwable, "Failed to load playlists");
        });
  }

  @Override public void bind(PlaylistDialogView view) {

    this.view = view;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/presenters/PlaylistDialogPresenterImpl.kt
package com.kelsos.mbrc.presenters

import com.kelsos.mbrc.extensions.task
import com.kelsos.mbrc.interactors.PlaylistInteractor
import com.kelsos.mbrc.ui.views.PlaylistDialogView
import timber.log.Timber
import javax.inject.Inject

class PlaylistDialogPresenterImpl
@Inject constructor(private var playlistInteractor: PlaylistInteractor) : PlaylistDialogPresenter {
  private var view: PlaylistDialogView? = null

  override fun load() {
    playlistInteractor.getUserPlaylists()
        .task()
        .subscribe({ view?.update(it) }, { Timber.v(it, "Failed to load playlists") })
  }

  override fun bind(view: PlaylistDialogView) {

    this.view = view
  }
}
